src/cli.rs

```diff
@@ -495,9 +495,13 @@ pub struct Opt {
     )]
     pub blame_format: String,
 
-    /// Separator between the commit metadata and code sections of a line of git blame output.
-    #[structopt(long = "blame-separator", default_value = "│")]
-    pub blame_separator: String,
+    /// Separator between the commit metadata and code sections of a line of git blame output. Contains
+    /// the line number by default. Possible values are "off" to disable line numbers or a format
+    /// string which may contain exactly one placeholder. These are: "{}" to display the line number
+    /// on every line; "{block}" to only show the line number when a new blame block starts; or
+    /// "{every-N}" to show the line number with every block and every N-th (modulo) line.
+    #[structopt(long = "blame-line-number-suffix", default_value = "│{:^4}│")]
+    pub blame_line_number_suffix: String,
```

Comment:
I'm thinking about how to increase consistency with other options (I think that with so many options, consistency is valuable if we are to retain some sanity). How about

- Call this `blame_separator_format`. All the other options that have `{placeholder}`s have a name like `*-format`.
- Support both `{n}` to mean: replace this placeholder with the current line number.  This gives consistency with `line-numbers-{left,right}-format`, sort of -- those are `nm` and `np` to distinguish the minus and plus line numbers.
- Is the empty placeholder `{}` natural, or should we not support that?
- Now, the way I'm thinking of it, `{block}` and `{every-N}` are "magical" in that they are not placeholders in the sense that it's used elsewhere, but rather directives saying when line numbers should be inserted. I'm wondering whether`{n:block}` and `{n:every-N}` are even more justifiable, meaning "insert line numbers, but subject to a certain policy".

I note that `hyperlinks-file-link-format` uses `{line}` to mean current line number. Possibly that should be changed to `{n}`, although the breaking change is probably not worth it. Maybe at least that one should have `{n}` support added and maybe `{line}` dropped from docs or mentioned as deprecated.

Reply:
Also
- Do you think we need the `off` special case? It's the same as `│` I believe. Perhaps, users who care are going to be just as capable of looking up the default and copying the `│` character, as they are of learning the `off` special case?

Reply:
Good idea, now using `{n}`, `{n:block}` and `{n:every-N}` and extended the format regex to support it. Still need to work around `StructOpt` displaying `{n}` as a newline. An empty `{} ` could also be supported, but that would make the help text more complicated.

And I think just writing `off` is easier than copy/pasting a unicode ` │`, plus it prevents mistaking it for a plain ascii `|`pipe and wondering why the output doesn't look as good as with line numbers.

Reply:
> And I think just writing `off` is easier than copy/pasting a unicode ` │`, plus it prevents mistaking it for a plain ascii `|` pipe and wondering why the output doesn't look as good as with line numbers.

OK, I'm convinced. What do you think about using `none` instead of `off`?  That would be consistent with `--syntax-theme=none`.